### DISCUSSION TITLE
ST7789V Driver

### DISCUSSION THREAD

**Discussion by shamirtahir50** [Bodmer/TFT_eSPI#2070] [Q&A - General] [answered]
Hi, 

I am using TFT024B045 which contains ST7789V, I am programmoing it using ESP32.

I am defining this in User_Setup.h

#define ESP32_PARALLEL

#define ST7789_DRIVER

// ESP32 pins used for the parallel interface TFT
#define TFT_CS   27  // Chip select control pin
#define TFT_DC   14  // Data Command control pin - must use a pin in the range 0-31
#define TFT_RST  26  // Reset pin

#define TFT_WR   12  // Write strobe control pin - must use a pin in the range 0-31
#define TFT_RD   13

#define TFT_D0   16  // Must use pins in the range 0-31 for the data bus
#define TFT_D1   4  // so a single register write sets/clears all bits
#define TFT_D2   23
#define TFT_D3   22
#define TFT_D4   21
#define TFT_D5   19
#define TFT_D6   18
#define TFT_D7   17

Is this the right way?
I need to change the pin configuration or not?
Do you have any documentation about that?

**Comment by shamirtahir50** [comment]
TFT024B045 Data Sheet
[TFT024B045-2.4inch-IPS-ST7789V-MCU 16bit.pdf](https://github.com/Bodmer/TFT_eSPI/files/9747135/TFT024B045-2.4inch-IPS-ST7789V-MCU.16bit.pdf)

ST7789V Data Sheet
[ST7789V_SPEC_V1.2.pdf](https://github.com/Bodmer/TFT_eSPI/files/9747140/ST7789V_SPEC_V1.2.pdf)

Pin Configuration is mentioned in the tft data sheet

**Comment by Bodmer** [comment, marked as answer]
The display module has a 16 bit parallel interface. The only processor in the library that supports 16 bit parallel interface is the RP2040.

It appears the IMx pins (that set the interface mode) are not accessible.

**Reply by shamirtahir50** [reply]
So, the pinout is the following,

![image](https://user-images.githubusercontent.com/66121435/195140779-91b66cc7-069e-46c0-8502-943c5fb8fcb8.png)

Should I define this way? 

#define TFT_CS 27 // Chip select control pin
#define TFT_DC 14 // Data Command control pin - must use a pin in the range 0-31
#define TFT_RST 26 // Reset pin

#define TFT_WR 12 // Write strobe control pin - must use a pin in the range 0-31
#define TFT_RD 13
// will add backlight anode here
#define TFT_D0 16 // Must use pins in the range 0-31 for the data bus
#define TFT_D1 4 // so a single register write sets/clears all bits
#define TFT_D2 23
#define TFT_D3 22
#define TFT_D4 21
#define TFT_D5 19
#define TFT_D6 18
#define TFT_D7 17

or this pinout

#define TFT_SPI_PORT 1

// Pins - the PIO will control MOSI, SCLK, and DC pins
// Any digital GPIO pins may be used
// #define TFT_MISO  -1 // MISO is not used or supported
#define TFT_MOSI  11
#define TFT_SCLK  10
#define TFT_CS     9  // Chip select control pin
#define TFT_DC     8  // Data Command control pin
#define TFT_RST   12  // Reset pin (could connect to Arduino RESET pin
#define TFT_BL    13  // LED backlight
#define TFT_BACKLIGHT_ON HIGH

My LCD data sheet does not mention SPI pins explicitly. 

In the second one, I do not know which pin will be connected from the LCD pinout to SCLK. And will it work with only 6 pins connected?

**Reply by Bodmer** [reply]
Neither will work. As previously stated, the display is for 16 bit parallel interface only, that means 16 data bus connections are required.

**Comment by UmerSoftwares** [comment]
Is there any wiring information to connect this LCD to Rasberry PI pico that has RP2040 ?

**Reply by Bodmer** [reply]
Here are the connections I have used with the RP2040:
https://github.com/Bodmer/TFT_eSPI/blob/master/User_Setups/Setup107_RP2040_ILI9341_16bit_parallel.h
